src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPMigrationService.kt

```diff
@@ -50,31 +49,37 @@ class CSIPMigrationService(
 
   override suspend fun migrateNomisEntity(context: MigrationContext<CSIPIdResponse>) {
     log.info("attempting to migrate $this")
-    val nomisCSIPId = context.body.csipId
+    val nomisCSIPReportId = context.body.csipId
     val migrationId = context.migrationId
 
-    csipMappingService.getCSIPReportByNomisId(nomisCSIPId)
+    csipMappingService.getCSIPReportByNomisId(nomisCSIPReportId)
       ?.run {
-        log.info("Will not migrate the CSIP $nomisCSIPId since it was already mapped to DPS CSIP ${this.dpsCSIPReportId} during migration ${this.label}")
+        log.info("Will not migrate the CSIP $nomisCSIPReportId since it was already mapped to DPS CSIP ${this.dpsCSIPReportId} during migration ${this.label}")
       }
       ?: run {
-        val nomisCSIPResponse = nomisApiService.getCSIP(nomisCSIPId)
+        val nomisCSIPResponse = nomisApiService.getCSIP(nomisCSIPReportId)
         csipService.migrateCSIP(nomisCSIPResponse.toDPSSyncRequest(actioned = nomisCSIPResponse.toActionDetails()))
-          .also { migratedCSIP ->
+          .also { syncResponse ->
             // At this point we need to determine all mappings and call the appropriate mapping endpoint
-            val dpsCSIPReportId = migratedCSIP.mappings.first { it.component == ResponseMapping.Component.RECORD }.uuid.toString()
-            createCSIPReportMapping(nomisCSIPId = nomisCSIPId, dpsCSIPId = dpsCSIPReportId, context = context)
-
-            // TODO **** MAP FACTORS ****
-            // TODO **** MAP ATTENDEES ****
-            // TODO **** MAP INTERVIEWS ****
-            // TODO **** MAP PLANS ****
-            // TODO **** MAP REVIEWS ****
-
+            val dpsCSIPReportId = syncResponse.filterReport().uuid.toString()
+            createMapping(
+              context,
+              CSIPFullMappingDto(
```

Comment:
All CSIP Mappings are now passed through as part of one call to the Mapping Service